dashboard/app/models/plc/course_unit.rb

```diff
@@ -34,22 +34,17 @@ def get_required_learning_module_ids
     plc_learning_modules.required.pluck(:id)
   end
 
-  def get_top_modules_of_each_type_from_user_selections(learning_module_ids)
+  def get_top_modules_of_each_type_from_user_selections(learning_module_ids_and_weights)
     # Preload all of the learning modules, that way we don't have to hit the DB multiple times
-    learning_module_ids.compact!
-
-    return [] if learning_module_ids.nil? || learning_module_ids.empty?
-    learning_module_map = Hash[*Plc::LearningModule.find(learning_module_ids).flat_map {|learning_module| [learning_module.id, learning_module]}]
-
-    #This will make a hash map of learning_module => number of times the user picked it in the evaluation, sorted with the
-    #most frequently selected ones first
-    searching_map = Hash[*learning_module_ids.group_by{ |v| v }.map{ |k, v| [learning_module_map[k.to_i], v.size] }.sort_by { |x| x[1] }.reverse.flatten]
+    return [] if learning_module_ids_and_weights.nil? || learning_module_ids_and_weights.empty?
```

Comment:
Oh thank heavens this exists.

I keep forgetting that nil is an object and not some scary thing like null in Java